### PR TITLE
SR-IOV: Adding SR-IOV

### PR DESCRIPTION
**Pull request by ntsbtz**
This commit support to configure single root I/O virtualization. Also the test has been added for the same.

### CODE REVIEW

File: include/network-config-manager.h.in [hunk 1]
```diff
@@ -114,6 +114,8 @@ int ncm_configure_link_coalesce(int argc, char *argv[]);
 int ncm_configure_link_coald_frames(int argc, char *argv[]);
 int ncm_configure_link_coal_pkt(int argc, char *argv[]);
 
+int ncm_configure_add_sr_iov(int argc, char *argv[]);
+
```
Comment (ssahani) [inline include/network-config-manager.h.in:118]: `ncm_configure_add_sr_iov` -> `ncm_configure_sr_iov`

File: src/manager/network-config-manager.c [hunk 2]
```diff
@@ -4759,6 +4760,162 @@ _public_ int ncm_configure_link_coal_pkt(int argc, char *argv[]) {
         return 0;
 }
 
+_public_ int ncm_configure_add_sr_iov(int argc, char *argv[]) {
+        _cleanup_(netdev_sriov_unrefp) SRIov *s = NULL;
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4764]: `ncm_configure_sr_iov`

Comment (ssahani) [inline src/manager/network-config-manager.c:4764]: `SRIov` -> `SRIOV`

File: src/manager/network-config-manager.c [hunk 3]
```diff
@@ -4759,6 +4760,162 @@ _public_ int ncm_configure_link_coal_pkt(int argc, char *argv[]) {
...
+                if (string_equal(argv[i], "vf")) {
+                        parse_next_arg(argv, argc, i);
+
+                        r = parse_uint32(argv[i], &v);
+                        if (r < 0) {
+                                log_warning("Failed to configure sriov vf ='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        s->vf = strdup(argv[i]);
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4793]:
```
   s->vf = strdup(argv[i]);
   if (! s->vf )
   return log_oom()
```

File: src/manager/network-config-manager.c [hunk 4]
```diff
@@ -4759,6 +4760,162 @@ _public_ int ncm_configure_link_coal_pkt(int argc, char *argv[]) {
...
+                if (string_equal(argv[i], "vlanid")) {
+                        parse_next_arg(argv, argc, i);
+
+                        r = parse_uint32(argv[i], &v);
+                        if (r < 0) {
+                                log_warning("Failed to configure sriov vlanid ='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        s->vlanid = strdup(argv[i]);
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4806]: `log_oom()`

File: src/manager/network-config-manager.c [hunk 5]
```diff
@@ -4759,6 +4760,162 @@ _public_ int ncm_configure_link_coal_pkt(int argc, char *argv[]) {
...
+
+                        if (!string_equal(argv[i], "auto")) {
+                                r = parse_boolean(argv[i]);
+                                if (r < 0) {
+                                        log_warning("Failed to parse sriov linkstate '%s': %s", argv[i], g_strerror(-r));
+                                        return r;
+                                }
+                        }
+
+                        s->linkstate = strdup(argv[i]);
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4883]: `log_oom()`

File: src/manager/network-sriov.h [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "alloc-util.h"
+#include "network-util.h"
+#include "config-file.h"
+
+typedef struct SRIov {
```
Comment (ssahani) [inline src/manager/network-sriov.h:11]: `SRIov` -> `SRIOV`

**Review by ssahani — CHANGES_REQUESTED** [review]
Looks good. Tiny changes required

File: include/network-config-manager.h.in (hunk 1 above)
Comment (ntsbtz) [reply include/network-config-manager.h.in:118]: Thanks @ssahani for review, And all the review comments has been taken care.

**Pull request merged**